src/System.Text.Json/tests/JsonDocumentDetachTests.cs

```diff
@@ -0,0 +1,344 @@
+
+            // After document Dispose
+            Assert.Equal("[]", detached.GetRawText());
+        }
+
+        [Fact]
+        public static void DetachInnerElementFromIsolatedDocument()
+        {
+            JsonElement detached;
+
+            using (JsonDocument doc = JsonDocument.Parse("[[]]"))
+            using (JsonDocument isolated = doc.Detach())
+            {
+                JsonElement inner = isolated.RootElement[0];
+                detached = inner.Detach();
+
+                Assert.True(inner.IsDetached, "inner.IsDetached");
+                Assert.True(detached.IsDetached, "detached.IsDetached");
+
+                Assert.Equal(inner.GetRawText(), detached.GetRawText());
+                Assert.NotSame(doc, detached.SniffDocument());
+            }
+
+            // After document Dispose
+            Assert.Equal("[]", detached.GetRawText());
+        }
+
+        [Fact]
+        public static void DetachInnerElementFromDetachedElement()
+        {
+            JsonElement detached;
+
+            using (JsonDocument doc = JsonDocument.Parse("[[[]]]"))
+            {
+                JsonElement middle = doc.RootElement[0].Detach();
+                JsonElement inner = middle[0];
+                detached = inner.Detach();
+
+                Assert.Equal(inner.GetRawText(), detached.GetRawText());
+                Assert.NotSame(doc, detached.SniffDocument());
+                Assert.NotSame(middle.SniffDocument(), detached.SniffDocument());
+                Assert.NotSame(inner.SniffDocument(), detached.SniffDocument());
+            }
+
+            // After document Dispose
+            Assert.Equal("[]", detached.GetRawText());
+        }
+
+        [Fact]
+        public static void DetachAtInnerNumber()
+        {
+            DetachAtInner("1.21e9", JsonValueType.Number);
+        }
+
+        [Fact]
+        public static void DetachAtInnerString()
+        {
+            DetachAtInner("\"  this  string  has  \\u0039 spaces\"", JsonValueType.String);
+        }
+
+        [Fact]
+        public static void DetachAtInnerTrue()
+        {
+            DetachAtInner("true", JsonValueType.True);
+        }
+
+        [Fact]
+        public static void DetachAtInnerFalse()
+        {
+            DetachAtInner("false", JsonValueType.False);
+        }
+
+        [Fact]
+        public static void DetachAtInnerNull()
+        {
+            DetachAtInner("null", JsonValueType.Null);
+        }
+
+        [Fact]
+        public static void DetachAtInnerObject()
+        {
+            DetachAtInner(
+                @"{
+  ""this"":
+  [
+    {
+      ""object"": 0,
+
+
+
+
+      ""has"": [ ""whitespace"" ]
+    }
+  ]
+}",
+                JsonValueType.Object);
+        }
+
+        [Fact]
+        public static void DetachAtInnerArray()
+        {
+            DetachAtInner(
+                @"[
+{
+  ""this"":
+  [
+    {
+      ""object"": 0,
+
+
+
+
+      ""has"": [ ""whitespace"" ]
+    }
+  ]
+},
+
+5
+
+,
```

Comment:
Why so much white space?

Reply:
Just making sure that nothing was depending on the input being in a certain pretty-printed way.

It was actually really hard to make my naturally very tidy mind write this.